src/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -1319,6 +1319,25 @@ public static void ConstrainedCopy(Array sourceArray, int sourceIndex, Array des
             Assert.Equal(expected, destinationArrayClone);
         }
 
+        [OuterLoop] // Allocates large array
+        [Fact]
+        public static void Copy_LargeMultiDimensionalArray()
+        {
+            // If this test is run in a 32-bit process, the large allocation will fail.
+            if (IntPtr.Size != sizeof(long))
+            {
+                return;
+            }
+
+            short[,] a = new short[2, 2_000_000_000];
```

Comment:
in future could this use the API in https://github.com/dotnet/corefx/issues/42293 ?

Reply:
It cannot. That API is for single dimensional array.